v2/clients/http/utils/post.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+)
+
+// Helper method to make the post JSON request and return the body
+func PostRequest(
+	ctx context.Context,
+	returnValuePointer interface{},
+	url string,
+	data interface{}) errors.EdgeX {
+
+	jasonEncodedData, err := json.Marshal(data)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to encode input data to JSON", err)
+	}
+
+	res, err := PostRequestWithURL(ctx, url, jasonEncodedData)
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	if err := json.Unmarshal(res, returnValuePointer); err != nil {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "failed to parse the response body", err)
+	}
+	return nil
+
+}
+
+// PostRequestWithURL will make a POST request to the specified URL.
+// It returns the body as a byte array if successful and an error otherwise.
+func PostRequestWithURL(ctx context.Context, url string, data []byte) ([]byte, errors.EdgeX) {
```

Comment:
Naming of these func needs tweaking. `PostRequestWithURL` and `PostRequest` both take urls. How about this one change to `PostRawDataRequest`

Reply:
Thanks @lenny-intel. `PostRawDataRequest` is really better than the old one.